### PR TITLE
xe: sdpa: Fix TL causal mask where Q greater than K and V

### PR DESCRIPTION
**Pull request by umar456**
This pull request introduces support for handling scenarios where the query (`Q`) dimension is larger than the key/value (`K/V`) dimensions in the context of causal masking for self-attention. The changes include updates to the kernel logic, new test cases, and additional test parameters to ensure coverage for this edge case.

### Kernel Logic Updates:
* Modified the `micro_sdpa` kernel in `src/gpu/intel/ocl/micro_sdpa.cl` to handle the `ATTN_MASK_TOP_LEFT` case by adjusting the calculation of `k0end` for causal masking. This ensures correct behavior when `Q` is larger than `K/V`.

### New Test Cases:
* Added a new regression test in `tests/benchdnn/inputs/graph/complex_fusion/harness_mha_all` to validate the behavior of causal masking when `Q` is larger than `K/V`. This test includes specific input shapes and attributes to simulate the edge case.

### Expanded Test Parameters:
* Updated the test suite in `tests/gtests/internals/test_sdpa.cpp` to include additional test cases for the `mask_type::causal_tl` and `mask_type::causal_br` scenarios.

Fixes: [MFDNN-13851](https://jira.devtools.intel.com/browse/MFDNN-13851)

### CODE REVIEW

**Review by syurkevi — APPROVED** [review]

**Comment by umar456** [comment]
make test
disable benchdnn_all
enable benchdnn_graph
enable test_device_gpu
enable arch_gpu_xe-hpc
enable arch_gpu_xe-hpg-atsm
enable arch_gpu_xe-hpg-dg2
enable arch_gpu_xe-lp
enable arch_gpu_xe-lpg
enable arch_gpu_xe-lpg+
enable arch_gpu_xe2-hpg-bmg
enable arch_gpu_xe2-lpg

File: tests/benchdnn/inputs/graph/complex_fusion/harness_mha_all [hunk 1]
```diff
@@ -154,3 +154,10 @@
 
 # regression tests
 --reset --dt=f32,bf16,f16 --in-shapes=0:1x8x4096x16+1:1x8x5x16+5:1x1x4096x5+8:1x8x5x16 --case=complex_fusion/mha/sdpa-plain-simplified-f16.json
+
+# MFDNN-13851: causal mask with Q larger than K/V
+--reset
+--dt=15:f32+16:f32+5:f32+21:f32
+--in-shapes=0:2x10x64x77*abdc+1:2x10x1x77+2:2x10x1x77+3:2x10x2304x64+6:2x10x77x64+7:2x10x77x1+8:2x10x77x1
+--op-attrs=0:group_shape:1x1x64x1+8:group_shape:1x1x1x64
+--case=complex_fusion/mha/sdpa-compressed-kv-implicit-causal-mask-int8-gs128.json
```
Comment (ElaineBao) [inline tests/benchdnn/inputs/graph/complex_fusion/harness_mha_all:163]: As the case in MFDNN-13851 is a floating-point case, could you please add case for floating-point? Thanks! You can use:

```
--reset --dt=1:bf16+2:bf16+3:bf16+6:bf16+104:bf16 --in-shapes=1:2x10x2304x64:acbd+2:2x10x77x64:acbd+5:2x10x77x64:acbd --case=complex_fusion/mha/sdpa-plain-implicit-causal-mask-fp32-bs1.json
--reset --dt=1:f16+2:f16+3:f16+6:f16+104:f16 --in-shapes=1:2x10x2304x64:acbd+2:2x10x77x64:acbd+5:2x10x77x64:acbd --case=complex_fusion/mha/sdpa-plain-implicit-causal-mask-fp32-bs1.json
--reset --in-shapes=1:2x10x2304x64:acbd+2:2x10x77x64:acbd+5:2x10x77x64:acbd --case=complex_fusion/mha/sdpa-plain-implicit-causal-mask-fp32-bs1.json
```

**Comment by umar456** [comment]
make test
disable benchdnn_all
enable benchdnn_graph
enable test_device_gpu
enable arch_gpu_xe-hpc
enable arch_gpu_xe-hpg-atsm
enable arch_gpu_xe-hpg-dg2
enable arch_gpu_xe-lp
enable arch_gpu_xe-lpg
enable arch_gpu_xe-lpg+
enable arch_gpu_xe2-hpg-bmg
enable arch_gpu_xe2-lpg

File: tests/benchdnn/inputs/graph/complex_fusion/harness_mha_all [hunk 2]
```diff
@@ -154,3 +154,14 @@
 
 # regression tests
 --reset --dt=f32,bf16,f16 --in-shapes=0:1x8x4096x16+1:1x8x5x16+5:1x1x4096x5+8:1x8x5x16 --case=complex_fusion/mha/sdpa-plain-simplified-f16.json
+
+# MFDNN-13851: causal mask with Q larger than K/V
```
Comment (dzarukin) [inline tests/benchdnn/inputs/graph/complex_fusion/harness_mha_all]:
```suggestion
# causal mask with Q larger than K/V
```

**Review by dzarukin — APPROVED** [review]
